src/syft/core/plan/translation/torchscript/plan_translate.py

```diff
@@ -0,0 +1,108 @@
+        We use nn.Module to store input kwarg names inside torchscript.
+        """
+
+        def __init__(self, kwarg_names: List[str]):
+            _kwarg_names: List[str]
+            super(PlanWrapper, self).__init__()
+            self._kwarg_names = kwarg_names
+
+        def forward(self, *args: Any) -> Any:
+            kwarg_ptrs: Dict[str, Pointer] = {}
+
+            # Since Syft Plan needs pointers as args,
+            # reverse-map actual arg values to pointers
+            for name, arg in zip(self._kwarg_names, args):
+                found = False
+                for ptr in ROOT_CLIENT.store:
+                    store_value: Any = PLAN_BUILDER_VM.store[ptr.id_at_location].data
+                    is_list = isinstance(arg, (SyList, list)) and isinstance(
+                        store_value, (SyList, list)
+                    )
+                    is_dict = isinstance(
+                        arg, (dict, Dict, OrderedDict, SyOrderedDict)
+                    ) and isinstance(
+                        store_value, (dict, Dict, OrderedDict, SyOrderedDict)
+                    )
+                    if is_list:
+                        # Assume lists match if their contents are equal
+                        found = all(map(lambda a, b: a is b, store_value, arg))
+                    elif is_dict:
+                        # Assume dicts match if their contents are equal
+                        arg_keys, store_value_keys = sorted(arg.keys()), sorted(
+                            store_value.keys()
+                        )
+                        arg_values = tuple(arg[k] for k in arg_keys)
+                        store_value_values = tuple(arg[k] for k in arg_keys)
+                        keys_match = all(
+                            map(lambda a, b: a == b, arg_keys, store_value_keys)
+                        )
+                        values_match = all(
+                            map(lambda a, b: a is b, arg_values, store_value_values)
+                        )
+                        found = keys_match and values_match
+                    else:
+                        found = store_value is arg
+                    if found:
+                        kwarg_ptrs[name] = ptr
+                        break
+
+                if not found:
+                    traceback_and_raise(f"Could not map '{name}' arg value to Pointer")
+
+            # Execute Plan in the same VM where it was built!
+            res = plan(PLAN_BUILDER_VM, PLAN_BUILDER_VM.verify_key, **kwarg_ptrs)
+            return res
+
+    # Builder VM holds inputs in the store, retrieve actual arg values from there
+    kwarg_names = list(plan.inputs.keys())
+    args = tuple(
+        PLAN_BUILDER_VM.store[plan.inputs[name].id_at_location].data
+        for name in kwarg_names
+    )
+
+    # Remove wrappers, th.jit.trace doesn't like them
+    args = tuple(list(arg) if isinstance(arg, SyList) else arg for arg in args)
```

Comment:
I think you can check for SyPrimitive and then use .downcast() on it? note sure.